array_api_tests/pytest_helpers.py

```diff
@@ -483,6 +487,48 @@ def assert_fill(
         assert xp.all(xp.equal(out, xp.asarray(fill_value, dtype=dtype))), msg
 
 
+def scalar_eq(s1: Scalar, s2: Scalar) -> bool:
+    if cmath.isnan(s1):
+        return cmath.isnan(s2)
+    else:
+        return s1 == s2
+
+
+def assert_kw_copy(func_name, x, out, data, copy):
+    """
+    Assert copy=True/False functionality is respected
+
+    TODO: we're not able to check scalars with this approach
```

Comment:
We should definitely be able to check 0-dimensional arrays. What we can't check is np.generic, whiich AFAIK is a numpy-specific type